src/inferFullTargets.ts

```diff
@@ -377,6 +380,7 @@ export function inferRangeEndTarget(
 
   const mark =
     target.mark ??
+    (target.selectionType === "token" ? CURSOR_MARK_TOKEN : null) ??
```

Comment:
This is not quite right. It won't work for "take second char trap past end of token", which should refer to the end of "token trap" rather than "token this", to be consistent with eg "take trap past end of line"

Reply:
I think combining sub token with a range is just incorrect. Sub token should be defined as a single token.

Reply:
- [ ] add test